gitless/tests/test_core.py

```diff
@@ -878,12 +878,20 @@ def setUp(self):
     """Creates temporary local Git repo to use as the remote."""
     super(TestRemote, self).setUp()
 
+    # Create a repo for including in remote as a submodule
+    self.submodule_repo_path = tempfile.mkdtemp(prefix='gl-remote-test-submodule')
+    os.chdir(self.submodule_repo_path)
+    submodule_repo = core.init_repository()
+    git.commit(allow_empty=True, m='Initialize submodule repository')
+
     # Create a repo to use as the remote
     self.remote_path = tempfile.mkdtemp(prefix='gl-remote-test')
     os.chdir(self.remote_path)
     remote_repo = core.init_repository()
     remote_repo.create_branch(
         REMOTE_BRANCH, remote_repo.revparse_single('HEAD'))
+    git.submodule('add', '-f', self.submodule_repo_path)
```

Comment:
The `-f` tries fixing [build][build] failed due to AppVeyor gitignore configuration:

```
ERROR: test_clone_from_remote
(gitless.tests.test_core.TestInitFromRemote)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\gitless\gitless\tests\test_core.py", line 893, in
setUp
    git.submodule('add', self.submodule_repo_path)
  File "C:\Python27\lib\site-packages\pbs.py", line 456, in __call__
    return RunningCommand(command_ran, process, call_args, actual_stdin)
  File "C:\Python27\lib\site-packages\pbs.py", line 168, in __init__
    self._handle_exit_code(self.process.wait())
  File "C:\Python27\lib\site-packages\pbs.py", line 235, in
_handle_exit_code
    raise get_rc_exc(rc)(self.command_ran, self._stdout, self._stderr)
ErrorReturnCode_1:
Ran: u'git submodule add
c:\\users\\appveyor\\appdata\\local\\temp\\1\\gl-remote-test-submodulest4nzk'
STDOUT:

STDERR:
  The following path is ignored by one of your .gitignore files:
\users\appveyor\appdata\local\temp\1\gl-remote-test-submodulest4nzk
Use -f if you really want to add it.
```

[build]: https://ci.appveyor.com/project/spderosso/gitless/build/1.0.38/job/8jrcs6sp23n39h94

Reply:
Don't you need to do the `git submodule add` in the original repo (after the line that does `os.chdir(self.path)`)?